packages/interactivity/src/proxies/test/context-proxy.ts

```diff
@@ -277,6 +277,30 @@ describe( 'Interactivity API', () => {
 					'fromFallback',
 				] );
 			} );
+
+			it( 'should handle deeply nested properties that are initially undefined', () => {
+				const context: any = proxifyContext( {}, {} );
+
+				let deepValue: any;
+				const spy = jest.fn( () => {
+					deepValue = context.a?.b?.c?.d;
+				} );
+				effect( spy );
+
+				// Initial call, the deep value is undefined
+				expect( spy ).toHaveBeenCalledTimes( 1 );
+				expect( deepValue ).toBeUndefined();
+
+				// Add a deeply nested object to the context
+				context.a = { b: { c: { d: 'test value' } } };
+
+				// The effect should be called again
+				expect( spy ).toHaveBeenCalledTimes( 2 );
+				expect( deepValue ).toBe( 'test value' );
```

Comment:
@DAreRodz both of those assertions fail: 

- `expect( spy ).toHaveBeenCalledTimes( 2 );`

   ```
    Expected number of calls: 2
    Received number of calls: 1
   ```

- `expect( deepValue ).toBe( 'test value' );`

   ```
    Expected: "test value"
    Received: undefined
   ```
  
But I'm not 100% sure if those are precisely the failures we wanted to see.

Reply:
Thanks for the PR, @michalczaplinski. 😁


> I'm not 100% sure if those are precisely the failures we wanted to see.

They are. The values received are the same as those returned in the initial call. Basically, the `spy` function has not been executed again because the signal for `context.a` has not been updated.

Reply:
I had to stare at the test a little longer to convince myself that it was actually the case 😅.

Excellent, I'll add the fix tomorrow 🙂